### PR TITLE
Fix isByteReg() assert for x86

### PR DESCRIPTION
**Pull request by kunalspathak**
The change is similar to the one done in https://github.com/dotnet/runtime/pull/46567. If the type if `BYTE`, the use byte regs.

Fixes the following jitstressregs failures on Windows x86

https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-47307-head-c9ab37658a344842be/System.Linq.Queryable.Tests/console.2ce37104.log?sv=2019-07-07&se=2021-03-10T08%3A39%3A45Z&sr=c&sp=rl&sig=GdS1KlEkJHVcKz7lwpZeszudWYokZpr%2Bt7IGv4c61co%3D

https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-47307-head-c9ab37658a344842be/System.Linq.Tests/console.4bc4fd10.log?sv=2019-07-07&se=2021-03-10T08%3A39%3A45Z&sr=c&sp=rl&sig=GdS1KlEkJHVcKz7lwpZeszudWYokZpr%2Bt7IGv4c61co%3D

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

File: src/coreclr/jit/lsrabuild.cpp [hunk 1]
```diff
@@ -3207,8 +3207,24 @@ void LinearScan::BuildStoreLocDef(GenTreeLclVarCommon* storeLoc,
...
         }
     }
-    RefPosition* def =
-        newRefPosition(varDefInterval, currentLoc + 1, RefTypeDef, storeLoc, allRegs(varDsc->TypeGet()), index);
+
+    regMaskTP defCandidates = RBM_NONE;
+    var_types type          = varDsc->TypeGet();
+
+#ifdef TARGET_X86
+    if (varTypeIsByte(type))
```
Comment (sandreenko) [inline src/coreclr/jit/lsrabuild.cpp:3215]: would it be correct to add this logic into `allRegs(type)` instead?

Comment (kunalspathak) [reply src/coreclr/jit/lsrabuild.cpp:3215]: That sounds to be appropriate fix, but currently there are too many usage of `allRegs(type)` and I am not sure how that suggestion would impact other flows. My recommendation is to get this fix and generalize it in future if we find more cases like this.

Comment (sandreenko) [reply src/coreclr/jit/lsrabuild.cpp:3215]: Could you please add an assert to `allRegs(type)` that `type != TYP_BYTE` to see if any other calls are impacted, if it fails then it should show other existing issues.

**Review by sandreenko — APPROVED** [review]
LGTM with a follow-up analysis.

**Comment by kunalspathak** [comment]
@sandreenko  - That sounds reasonable. Let me try that out and run jitstressregs jobs.

**Comment by kunalspathak** [comment]
> @sandreenko - That sounds reasonable. Let me try that out and run jitstressregs jobs.

Looks like we hit the assert while compiling System.Private.CoreLib.dll so looks like we need more deeper analysis on right place to generalize it. [Here](https://dev.azure.com/dnceng/public/_build/results?buildId=1020218&view=results) are the failures. I will revert the assert and merge this PR.

**Pull request merged**